src/common/StreamingFactoryBase.sol

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.19;
+
+import {CREATE3} from "solmate/utils/CREATE3.sol";
+import {Ownable} from "openzeppelin-contracts/access/Ownable.sol";
+import {IERC20} from "openzeppelin-contracts/token/ERC20/IERC20.sol";
+
+import {AddressZero, AlreadyDeployed} from "./Errors.sol";
+
+/**
+ * @title StreamingFactoryBase
+ * @dev A base contract for creating and managing streaming contract instances.
+ */
+abstract contract StreamingFactoryBase is Ownable {
+    /// @dev The addresses of deployed instances.
+    address[] public deployedAddresses;
+    /// @dev The number of deployed instances.
+    uint256 public deployedCount;
+
+    /// @dev The address of the owner of the deployed instances.
+    address public instanceOwner;
+
+    event Deployed(address indexed vault, address indexed deployed);
+    event InstanceOwnerUpdated(address indexed sender, address oldOwner, address newOwner);
+
+    constructor(address _instanceOwner) Ownable(msg.sender) {
+        _checkZeroAddress(_instanceOwner);
+
+        instanceOwner = _instanceOwner;
+    }
+
+    /**
+     * @dev Sets the address of the owner of the deployed YieldStreaming instances.
+     * @param _newInstanceOwner The address of the new owner of the deployed YieldStreaming instances.
+     */
+    function setInstanceOwner(address _newInstanceOwner) public onlyOwner {
```

Comment:
is there a way to design the yield contract without any need for an owner? i know its for setting some tolerance but it seems like a liability

Reply:
tbh I don't see a way to avoid it except to hardcode that tolerance value. Why do you see it as a liability? What's the worst thing that can happen because of it?

Reply:
with owners we can't offer it as a neutral tool/wrapper over erc4626. with a factory deploy, anyone can deploy and make themselves owner, then later act maliciously. and over time we would have a web of owners and would have to upkeep some whitelist of good owners and bad and we cannot just trust the factory deploy. either that or we would have to permission the deploy which limits the utility. 

it would be more preferable if anyone could just "i want to stream yield on this vault not currently listed": deploy a streamer, set up streams. then later anyone can also use that streamer and we can list it without having to authenticate the owner. then we can just trust the code, no?

Reply:
atm, the factory also has an owner so it's not true that anyone can deploy/create new instances. If we decide to hardcode the "loss tolerance" to for example 2%, we can get rid of ownership of both factory and streaming contracts.

Reply:
or just remove it as a global setting all together and have it as a optional parameter when opening new streams. so i can select i am comfortable with 2% loss for example and more then that i want it to revert. comparable to a slippage parameter

Reply:
that's a great idea, thx :) I'll make the changes in a separate PR. I'm going to merge this one and reference this conversation in that new PR